modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SqlSchemaManagerImpl.java

```diff
@@ -64,6 +76,30 @@ public SchemaPlus schema(@Nullable String schema) {
         return schema != null ? calciteSchema.getSubSchema(schema) : calciteSchema;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    @NotNull
+    public InternalIgniteTable tableById(IgniteUuid id) {
+        ensureTableUpToDate(id);
+
+        InternalIgniteTable table = tablesById.get(id);
```

Comment:
what the goal of _tablesById_ structure ? i found 3 usages and 2 of them in listeners, u can easily return table from _ensureTableUpToDate_ isn`t it ?

Reply:
The `tablesById` keeps tables in terms of SQL engine, whilst `ensureTableUpToDate` could either returns it in terms of `TableManager` or wraps it on every call.

Reply:
ok, now it`s clear to me, possibly we need to swap them ?
additionally - do i correctly understand that this issue only about waiting of creation i suppose still possible to drop table while some queries are using it ? and all these stuff is unsafe for now ? isn`t it ?

Reply:
That is true this patch only about races between initial table creation and querying this table. The problem of concurrent table modification (as well as deletion) should be addressed on a higher level, because there is a dozen of places which expect the table to not be modified along the query planning/execution.

Reply:
can we swap:
1.  InternalIgniteTable table = tablesById.get(id);
2. if (table == null) {
ensureTableUpToDate(id);
table = tablesById.get(id);
}

something like that ?

Reply:
Oh, now I got it! Fixed